WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCard.swift

```diff
@@ -46,13 +78,112 @@ private extension ReviewsDashboardCard {
             .disabled(viewModel.syncingData)
         }
     }
+
+    var reviewsFilterBar: some View {
+        HStack {
+            AdaptiveStack(horizontalAlignment: .leading) {
+                Text(Localization.status)
+                    .foregroundStyle(Color(.text))
+                    .subheadlineStyle()
+
+                Text("All") // TODO: dynamically change based on filter selection
+                    .subheadlineStyle()
+            }
+            Spacer()
+
+            Menu {
+                ForEach(viewModel.filters, id: \.self) { filter in
+                    Button {
+                        // TODO
+                    } label: {
+                        SelectableItemRow(title: filter.title, selected: false)
+                    }
+                }
+            } label: {
+                Image(systemName: "line.3.horizontal.decrease")
+                    .foregroundStyle(Color.secondary)
+            }
+        }
+    }
+
+    func ReviewRow(for review: ProductReview, isLastItem: Bool) -> some View {
+        HStack(alignment: .top) {
+            Image(systemName: "bubble.fill")
+                .foregroundStyle(review.status == .hold ? Color.secondary : Color(.wooCommercePurple(.shade60)))
+                .padding(.horizontal, Layout.padding)
+                .padding(.vertical, Layout.cardPadding)
+
+
+            VStack(alignment: .leading) {
+                // TODO: use actual product name
+                authorText(author: review.reviewer, productName: "Fallen Angel Candelabra")
+                    .bodyStyle()
+                    .padding(.trailing, Layout.padding)
+                reviewText(text: review.review, shouldDisplayStatus: review.status == .hold)
+                    .lineLimit(2)
+                    .subheadlineStyle()
+                    .padding(.trailing, Layout.padding)
+                    .renderedIf(review.review.isNotEmpty)
+                HStack(spacing: Layout.starRatingSpacing) {
+                    ForEach(0..<abs(review.rating), id: \.self) { _ in
```

Comment:
`abs` just in case somehow the value is negative and cause crash.

Reply:
You should put this code in an `if` check to render the stars only when the rating is larger than 0. It doesn't make sense to use `abs` in this case.

Reply:
Good point, that simplifies the logic too.

Added in https://github.com/woocommerce/woocommerce-ios/pull/12731/commits/73df51b708c06b59e6cd1aa6d3c7a9865d1cf180